test_simple_pe/test_simple_pe_magma.py

```diff
@@ -0,0 +1,66 @@
+from simple_pe.simple_pe_magma import define_pe
+import fault.random
+import operator
+import magma as m
+import mantle
+import os
+
+
+def test_simple_pe():
+    ops = [operator.add, operator.sub, operator.and_, operator.or_]
+    pe = define_pe(ops, T=m.UInt, data_width=16)
+
+    tester = fault.Tester(pe)
+
+    # Sanity check each op with random value
+    for i, op in enumerate(ops):
+        tester.poke(pe.opcode, i)
+        I0 = fault.random.random_bv(16)
+        I1 = fault.random.random_bv(16)
+        tester.poke(pe.I0, I0)
+        tester.poke(pe.I1, I1)
+        tester.eval()
+        tester.expect(pe.O, op(I0, I1))
+    tester.compile_and_run(target="coreir")
+
+    m.compile("test_simple_pe/build/pe", pe, output="coreir")
+    assert not os.system("coreir -i test_simple_pe/build/pe.json "
```

Comment:
are these exposed through `pycoreir`?

Reply:
Done, added an argument to `magma.compile` that introduces support for a `passes` list that will be a list of passes to run when compiling.